pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/URLRegexLookupProxyHandler.java

```diff
@@ -0,0 +1,113 @@
+
+    private static final Logger log = LoggerFactory.getLogger(URLRegexLookupProxyHandler.class);
+
+    private String regex;
+
+    private String replacement;
+
+    @Override
+    public void initialize(ProxyService proxy, ProxyConnection proxyConnection) {
+        super.initialize(proxy, proxyConnection);
+        this.regex = proxy.getConfiguration().getProperties().getProperty("urlRegexLookupProxyHandlerRegex");
+        if (Strings.isNullOrEmpty(this.regex)) {
+            throw new IllegalArgumentException("urlRegexLookupProxyHandlerRegex is not set");
+        }
+        this.replacement =
+            proxy.getConfiguration().getProperties().getProperty("urlRegexLookupProxyHandlerReplacement");
+        if (Strings.isNullOrEmpty(this.replacement)) {
+            throw new IllegalArgumentException("urlRegexLookupProxyHandlerReplacement is not set");
+        }
+    }
+
+    @Override
+    public void handleLookup(CommandLookupTopic lookup) {
+        if (log.isDebugEnabled()) {
+            log.debug("Received Lookup from {}", clientAddress);
+        }
+        long clientRequestId = lookup.getRequestId();
+        if (lookupRequestSemaphore.tryAcquire()) {
+            try {
+                LOOKUP_REQUESTS.inc();
+                String serviceUrl = getBrokerServiceUrl(clientRequestId);
+                if (serviceUrl != null) {
+                    if (lookup.isAuthoritative()) {
+                        performLookup(clientRequestId, lookup.getTopic(), serviceUrl, false, 10)
+                            .whenComplete(
+                                (brokerUrl, ex) -> {
+                                    if (ex != null) {
+                                        ServerError serverError = ex instanceof LookupException
+                                            ? ((LookupException) ex).getServerError()
+                                            : getServerError(ex);
+                                        proxyConnection.ctx().writeAndFlush(
+                                            Commands.newLookupErrorResponse(serverError, ex.getMessage(),
+                                                clientRequestId));
+                                    } else {
+                                        proxyConnection.ctx().writeAndFlush(
+                                            Commands.newLookupResponse(brokerUrl, brokerUrl, true,
+                                                CommandLookupTopicResponse.LookupType.Connect, clientRequestId,
+                                                true /* this is coming from proxy */));
+                                    }
+                                });
+                    } else {
+                        performLookup(clientRequestId, lookup.getTopic(), serviceUrl, false, 10)
+                            .whenComplete(
+                                (brokerUrl, ex) -> {
+                                    if (brokerUrl.matches(regex)) {
```

Comment:
we need to catch IAE in case the regex is malconfigured otherwise no error is displayed and the connection just times out

Reply:
done